src/main/java/seedu/address/model/task/Recurring.java

```diff
@@ -61,22 +64,51 @@ public RecurType parseRecurType(String recurringString) throws ParseException {
      * @param taskToReset
      * @return copied task with done set to undone
      */
-    public Task resetDone(Task taskToReset) {
-        assert taskToReset != null;
+    public Done resetDone(Done done) {
+        if (done.getIsDone()) {
+            done = new Done("N");
+        }
+        return done;
+    }
+
+    public Optional<Reminder> resetReminder(Optional<Reminder> currentOptReminder) {
```

Comment:
Just some nitpicking but it looks a little arrow-headed. Might it be possible to make it less so? Less nested ifs statements?

Reply:
Ok sure